fdbclient/include/fdbclient/TenantManagement.actor.h

```diff
@@ -215,11 +223,24 @@ createTenantTransaction(Transaction tr, TenantMapEntry tenantEntry, ClusterType
 }
 
 ACTOR template <class Transaction>
-Future<int64_t> getNextTenantId(Transaction tr) {
-	Optional<int64_t> lastId = wait(TenantMetadata::lastTenantId().get(tr));
-	int64_t tenantId = lastId.orDefault(-1) + 1;
-	if (BUGGIFY) {
+Future<int64_t> getNextTenantId(Transaction tr, int numTenantsCreating) {
+	state Optional<int64_t> lastId = wait(TenantMetadata::lastTenantId().get(tr));
+	if (!lastId.present()) {
+		// If the last tenant id is not present fetch the tenantIdPrefix (if any) and initalize the lastId
+		int64_t tenantIdPrefix =
+		    wait(KeyBackedProperty<int64_t>(TenantAPI::tenantIdPrefixKey).getD(tr, Snapshot::False, 0));
+		// Shift by 6 bytes to make the prefix the first two bytes of the tenant id
+		lastId = tenantIdPrefix << 48;
+	}
+	int64_t tenantId = lastId.get() + 1;
+	// Only increment the tenant Id by a large factor if there is enough tenantIds for all tenants being created
+	// (without overriding the tenant prefix)
+	if (BUGGIFY && tenantId < getMaxAllowableTenantId(lastId.get()) - numTenantsCreating) {
```

Comment:
Do we need this restriction now? If we try to allocate some tenant IDs and go over, then this would just cause us to exceed our limit by some amount and then hopefully we would get the `cluster_no_capacity` error.